### PR TITLE
PPTP-1699: Add backwards-look test yes-no page to registration

### PR DESCRIPTION
**Pull request by lmcewanhmrc**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.
 - [ ] No Accessibility errors/warnings reported by Wave

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoController.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  def nextPage(alreadyExceeded: Boolean): Call =
+    if (alreadyExceeded)
+      // todo should be page where we ask for date user exceeded
+      routes.LiabilityStartDateController.displayPage()
+    else
+      // Haven't already exceeded so ask if they will exceed next
+      routes.ExpectToExceedThresholdWeightController.displayPage()
+
+  def onSuccess(alreadyExceeded: Boolean)(implicit request: JourneyRequest[_]): Future[Result] = {
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoController.scala:70]: if we want to keep these methods broken out for readability, can we make them private.
however I wouldnt be opposed to putting them back together a bit, personally i find this less readable

File: build.sbt [hunk 2]
```diff
@@ -20,8 +20,8 @@ lazy val microservice = Project(appName, file("."))
       "uk.gov.hmrc.plasticpackagingtax.registration.views.html.components._",
       "uk.gov.hmrc.plasticpackagingtax.registration.views.components.Styles._"),
     // auto format code following .scalafmt.conf
-    scalafmtOnCompile in Compile := true,
-    scalafmtOnCompile in Test := true,
+//    scalafmtOnCompile in Compile := true,
```
Comment (liampan) [inline build.sbt:23]: can we have this back?

**Review by liampan — CHANGES_REQUESTED** [review]
Careful of the messages file apostrophe magic.

couple of optional comments also :)

File: test/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/ExceededThresholdYesNoViewSpec.scala [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      view must containElementWithID("answer-2")
+      view.getElementsByClass("govuk-label").get(1).text() mustBe "No"
+    }
+
+    "display 'Save and continue' button" in {
+      view must containElementWithID("submit")
+      view.getElementById("submit").text() mustBe "Save and continue"
+    }
+
+    // TODO need a working way to test which radio is checked (or neither) when form is first displayed
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/ExceededThresholdYesNoViewSpec.scala:73]: Is it not possible to create the form and call `fill()` and then check for the presence of `checked` attributes?

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
Couple of minor observations that are probably worth fixing.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoController.scala (hunk 1 above)
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoController.scala:70]: Done, thanks

File: build.sbt (hunk 2 above)
Comment (lmcewanhmrc) [reply build.sbt:23]: (We had a huddle about this)

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoController.scala [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  def displayPage(): Action[AnyContent] =
+    (authenticate andThen journeyAction) { implicit request =>
+      request.registration.liabilityDetails.exceededThresholdWeight match {
+        case Some(data) => Ok(page(ExceededThresholdYesNo.form().fill(data)))
+        case _          => Ok(page(ExceededThresholdYesNo.form()))
+      }
+    }
+
+  def hasErrors(form: Form[Boolean])(implicit request: Request[_]): Future[Result] =
+    Future.successful(BadRequest(page(form)))
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoController.scala:51]: Ok, yeah - thanks

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+            withLiabilityDetails(LiabilityDetails(exceededThresholdWeight = Some(true)))
+          )
+        authorizedUser()
+        mockRegistrationFind(registration)
+        val result = controller.displayPage()(getRequest())
+        status(result) mustBe Status.OK
+      }
+
+      "populate the form with the previous answer" in {
+        clearInvocations(mockPage) // todo oh dear move to before hook
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala:79]: I'm not sure... started to discuss some options, then Friday's bombshell hit... so it seemed less important

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+        authorizedUser()
+        val existingRegistration = mock[Registration](RETURNS_DEEP_STUBS)
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala:82]: Yeah, I noticed that! Mocking it allows us to test that the correct field was used to populate the form.

File: test/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/ExceededThresholdYesNoViewSpec.scala (hunk 3 above)
Comment (lmcewanhmrc) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/views/liability/ExceededThresholdYesNoViewSpec.scala:73]: I think so. Was adding a test along those lines when Friday's bombshell hit, so decided to just stick with the "standard" tests, at least for now

File: conf/messages.en [hunk 7]
```diff
@@ -412,11 +412,17 @@ liabilityExpectToExceedThresholdWeightPage.guidance.href = https://www.gov.uk/gu
 liabilityExpectToExceedThresholdWeightPage.radio.option.yes= Yes
 liabilityExpectToExceedThresholdWeightPage.radio.option.no= No
 
+liability.exceededThresholdWeight.sectionHeader = Liability details
+liability.exceededThresholdWeight.title = Has your organisation or group processed 10,000kg or more of plastic packaging since 1 April 2022?
+liability.exceededThresholdWeight.hint = If you're registering as a group, each member must have met this threshold.
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply conf/messages.en]: :) thanks

File: test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+      "return 200" in {
+        authorizedUser()
+        val result: Future[Result] = controller.displayPage()(getRequest())
+        status(result) shouldEqual Status.OK
+      }
+    }
+
+    "continuing an existing registration" should {
+
+      "user is authorised, a registration already exists and display page method is invoked" in {
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala:67]: Oops, yeah. I think that test has been superseded by the one after... checking

[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/liability/ExceededThresholdYesNoControllerSpec.scala:67]: It is redundant I reckon, removed it

**Review by RyanKellyHMRC — APPROVED** [review]

**Review by liampan — APPROVED** [review]

**Pull request merged**